### PR TITLE
docs: Add note about zero width unicode characters (#2139)

### LINKED ISSUE #2139: First line is not parsed as markdown
**Marked version:** 2.1.3

**Describe the bug**
I'm using `marked` in my `next.js` app. I have this code:

```
const fs = require('fs');
const marked = require("marked");

var content = fs.readFileSync('/sample/markdown/file.md', 'utf8');
content = marked(content);
```

And this is my `file.md`:

```
# Formal sciences

***

## Tools

- [GeoGebra](https://www.geogebra.org/)
```

And this is my output:

imageedit_1_3630574482

But if I only add one line to the beginning of my `file.md`, then I get this result:

imageedit_3_9689168460

**To Reproduce**

It works on your demo playground. But it doesn't work in my `next.js` app. And I only call your `marked` method passing my file's content. Nothing more.

- Create a `next.js` app
- Add `marked`
- Create an `html` file and serve it.
- Now change the extension of that `HTML` file to `.md`.
- First line is not respected as markdown.

If we create `.md` file in the first place, things work as expected. But when we change the extension of an existing HTML file to .md, first line breaks.

**Expected behavior**
The first line should be respected as markdown too. I can't add an empty line to all of my files.

### PR DESCRIPTION
**Pull request by matijagaspar**
## Description

- Fixes #2139

## What was attempted

Added clarification about ZERO WIDTH unicode characters interfering with parsing. 

## Contributor

- [x] no tests required for this PR.

## Committer

In most cases, this should be a different person than the contributor.

- [ ] CI is green (no forced merge required).
- [ ] Squash and Merge PR following [conventional commit guidelines](https://www.conventionalcommits.org/).

### CODE REVIEW

**Review by UziTech — APPROVED** [review]
Thanks! :100:

File: docs/INDEX.md [hunk 1]
```diff
@@ -34,6 +34,8 @@ npm install @types/marked # For TypeScript projects
 DOMPurify.sanitize(marked.parse(`<img src="x" onerror="alert('not happening')">`));
 ```
 
+**⚠️ Input: special ZERO WIDTH unicode characters (for example `\uFEFF`) might interfere with parsing. Some text editors add them at the start of the file (see: [#2139](https://github.com/markedjs/marked/issues/2139)).**
+
```
Comment (styfle) [inline docs/INDEX.md:38]: Let’s add an example below (like we do for sanitizing)

Comment (matijagaspar) [reply docs/INDEX.md:38]: Amended the commit

**Review by styfle — APPROVED** [review]
Thanks!

**Pull request merged**